Diagnostics/Get-NetView.PS1

```diff
@@ -1556,23 +1554,42 @@ function Environment {
     }
 } # Environment()
 
-function EnvDestroy {
+function EnvSetup {
     [CmdletBinding()]
     Param(
-        [parameter(Mandatory=$true)] [String] $OutDir
+        [parameter(Mandatory=$false)] [String] $OutputDirectory,
+        [parameter(Mandatory=$true)] [String] $WorkingDirectoryName
     )
-    If (Test-Path $OutDir) {
-        Remove-Item $OutDir -Recurse
+
+    # Process/Validate $OutputDirectory
+    function NormalizeOutDir {
+        if ($OutputDirectory) {
+            if (Test-Path $OutputDirectory) {
+                return (Resolve-Path $OutputDirectory).Path # full path
+            } else {
+                Throw "Get-NetView: The directory '$OutputDirectory' does not exist."
+            }
+        } # else $OutputDirectory is undefined
+        return [Environment]::GetFolderPath("Desktop") + "\" 
+    } # NormalizeOutDir ()
+
+    # Calculate working directory
+    $workDir = Join-Path $(NormalizeOutDir) $WorkingDirectoryName
+
+    # Delete working directory (and all of its contents) if it already exists
+    If (Test-Path $workDir) {
+        Remove-Item $workDir -Recurse
```

Comment:
The problem is that `Remove-Item -Recurse` will indiscriminately remove a directory and all its contents. Not an issue before because the output dir was hardcoded to the Desktop. Imagine if something like `%USERPROFILE%\Documents` were passed to `EnvDestroy`? My logic is that putting input validation and execution in the same step reduces the likelihood of bugs leading to data loss.

I'm not convinced that that `EnvSetup` is against cohesion because `EnvDestroy` and `EnvCreate` were overly simple in function and heavily dependent on one-another. There's never a reason to call one "by itself", i.e. without calling the other before or afterhand.

Why is a nested function not ideal? (a lambda would be better here, however) In this case, I think it makes slightly easier to see what the input to `Join-Path` is going to be.

Also, why is documenting code with comments worse than with function declarations?